README.md

```diff
@@ -72,48 +72,111 @@ This is a tutorial project of [Pocket Flow](https://github.com/The-Pocket/Pocket
 
 ## 🚀 Getting Started
 
-1. Clone this repository
+1. **Clone this repository**
+   ```bash
+   git clone https://github.com/The-Pocket/Tutorial-Codebase-Knowledge.git
+   cd Tutorial-Codebase-Knowledge
+   ```
 
-2. Install dependencies:
+2. **Set up a virtual environment (recommended)**
    ```bash
-   pip install -r requirements.txt
+   python -m venv venv
+   source venv/bin/activate  # On Windows: venv\Scripts\activate
    ```
 
-3. Set up LLM in [`utils/call_llm.py`](./utils/call_llm.py) by providing credentials. By default, you can use the AI Studio key with this client for Gemini Pro 2.5:
+3. **Install dependencies**
+   ```bash
+   pip install -r requirements.txt
+   ```
 
-   ```python
-   client = genai.Client(
-     api_key=os.getenv("GEMINI_API_KEY", "your-api_key"),
-   )
+4. **Configure LLM access**
+   
+   The tool supports multiple LLM providers. Configure at least one:
+   
+   - **Google Gemini (default)**: 
+     ```bash
+     # For Vertex AI:
+     export GEMINI_PROJECT_ID="your-project-id"
+     export GEMINI_LOCATION="us-central1"
+     # OR for AI Studio:
+     export GEMINI_API_KEY="your-api-key"
+     ```
+   
+   - **Anthropic Claude**:
+     ```bash
+     export ANTHROPIC_API_KEY="your-api-key"
+     # Uncomment Claude function in utils/call_llm.py
+     ```
+     
+   - **OpenAI**:
+     ```bash
+     export OPENAI_API_KEY="your-api-key"
+     # Uncomment OpenAI function in utils/call_llm.py
+     ```
+
+5. **Set up GitHub token (recommended)**
+   ```bash
+   export GITHUB_TOKEN="your-github-token"
    ```
 
-   You can use your own models. We highly recommend the latest models with thinking capabilities (Claude 3.7 with thinking, O1). You can verify that it is correctly set up by running:
+6. **Verify your setup**
    ```bash
    python utils/call_llm.py
    ```
 
-4. Generate a complete codebase tutorial by running the main script:
-    ```bash
-    # Analyze a GitHub repository
-    python main.py --repo https://github.com/username/repo --include "*.py" "*.js" --exclude "tests/*" --max-size 50000
+7. **Generate a tutorial**
+   ```bash
+   # From a GitHub repository
+   python main.py --repo https://github.com/username/repo --include "*.py" "*.js"
+   
+   # Or from a local directory
+   python main.py --dir /path/to/your/codebase --include "*.py"
+   ```
 
-    # Or, analyze a local directory
```

Comment:
could you keep these just in readme? I love the detailed instructions! But having two files may be confusing. If the worry is that there are too many texts, we can try  collapsible section in markdown to hide texts? https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab